### PR TITLE
NO_TICKET. Registering a health check callback now will not lead to a data race

### PR DESCRIPTION
**Pull request by bullgare**
API was not changed.

### CODE REVIEW

**Comment by bullgare** [comment]
Sure.
I just forgot it. Thanks, @dh-saurabh !

You can register new callbacks in any time.
And while registering a callback, you change a map of these callbacks (or structs, does not matter).
If at the same time an HTTP server already started, it's healthcheck handler reads values from the exact same map in another goroutine (as http handler is always running in another goroutine).

Most of the time that is not a problem, but in theory you can run into race condition with this map.

File: fdhttp/fdhandler/health_check.go [hunk 1]
```diff
@@ -105,7 +106,9 @@ func (h *HealthCheck) Init(r *fdhttp.Router) {
 
 // Register a new healthcheck Service.
 func (h *HealthCheck) Register(name string, s HealthChecker) {
+	h.servicesGuard.Lock()
```
Comment (dh-saurabh) [inline fdhttp/fdhandler/health_check.go]: How about using the `defer h.servicesGuard.Unlock()` ? 
The final code will look like this:
```
h.servicesGuard.Lock()
defer h.servicesGuard.Unlock()
h.services[name] = s
```
wdyt ?

Comment (bullgare) [reply fdhttp/fdhandler/health_check.go]: defer (in theory) adds more complexity and reduces the speed of the program.
I'm pretty sure that in this situation compiler will decide to make an optimization so there will be no differences.
But if someone will add more functionality to this function, we will have a bigger lock. Which will be at least unnecessary and can even lead to some deadlocks (again, in theory, and if we're really unlucky).
So, it's a pretty straightforward case, no multiple returns. And I decided to just stick to a simpler solution.
WDYT?

Comment (dh-saurabh) [reply fdhttp/fdhandler/health_check.go]: I think this is a simplest use case to use the `defer` it doesnt add any complexity unless:
- defer evaluation is resulted in `nil`
- using it inside a loop
etc etc
But sure keeping it simple is also important.

**Pull request merged**